src/utils/conda_environment_file.rs

```diff
@@ -2,22 +2,22 @@ use itertools::Itertools;
 use miette::IntoDiagnostic;
 use rattler_conda_types::ParseStrictness::Lenient;
 use rattler_conda_types::{Channel, MatchSpec};
-use serde::Deserialize;
+use serde::{Deserialize, Serialize};
 use std::str::FromStr;
 use std::{io::BufRead, path::Path, sync::Arc};
 
 use crate::config::Config;
 
-#[derive(Deserialize, Debug, Clone)]
+#[derive(Serialize, Deserialize, Debug, Clone)]
 pub struct CondaEnvFile {
     #[serde(default)]
-    name: Option<String>,
+    pub name: Option<String>,
     #[serde(default)]
-    channels: Vec<String>,
-    dependencies: Vec<CondaEnvDep>,
+    pub channels: Vec<String>,
+    pub dependencies: Vec<CondaEnvDep>,
 }
 
-#[derive(Deserialize, Debug, Clone)]
+#[derive(Serialize, Deserialize, Debug, Clone)]
 #[serde(untagged)]
 pub enum CondaEnvDep {
```

Comment:
I think if you implement a `impl From<MatchSpec> for CondaEnvDep` it should be pretty easy to add them. 

You already have the `PacakgeName` and `NamelessMatchSpec` in the manifest. 

So it should be easy to create a `MatchSpec` and use the `.to_string()` to get the actual string. e.g.:
```rs
let spec = MatchSpec::from_nameless(nameless_spec, Some(package_name));
```

Reply:
I'm able to quickly get some of the specs to render ok with `CondaEnvDep::Conda(format!("{}{}", name.as_source(), spec))`, but a handful (`*`) aren't looking right.

For pixi's `pixi.toml`
```yml
name: pixi-default-osx-arm64
channels:
- conda-forge
dependencies:
- pre-commit~=3.3.0
- rust~=1.77.0
- openssl3.*
- pkg-config0.29.*
- git2.42.0.*
- cffconvert>=2.0.0,<2.1
- tbump>=6.9.0,<6.10
```

I haven't tried writing an `impl` yet.

Reply:
It needs a `space`. But this is indeed basically it.

Reply:
Without an `impl` that gets me closer `CondaEnvDep::Conda(MatchSpec::from_nameless(spec, Some(name)).to_string())`:

```yml
name: pixi-default-osx-arm64
channels:
- conda-forge
dependencies:
- pre-commit ~=3.3.0
- rust ~=1.77.0
- openssl 3.*
- pkg-config 0.29.*
- git 2.42.0.*
- cffconvert >=2.0.0,<2.1
- tbump >=6.9.0,<6.10
```

Reply:
Wait, if there is a space and no leading operator, does that get interpreted as `=`?